guide_drafts/migration_guide.md

```diff
@@ -0,0 +1,277 @@
+# Diesel 2.0 Migration guide
+
+Diesel 2.0 changes large parts of the internal implementation of diesel. 
+In some cases this impacts code written using Diesel 1.4.x. 
+This document outlines notable changes and presents potential update strategies. 
+We recommend to start the upgrade by removing the usage of all items that 
+are deprecated on Diesel 1.4.x.
+
+We expect that not all users are effected equally by different required changes. 
+Any code base using migrating to Diesel 2.0 is expected to be affected at least by 
+the following changes:
```

Comment:
"The following changes are guaranteed to be affecting all of the users intending to migrate to Diesel 2.0:" ❔